secrethub/resource_access_rule.go

```diff
@@ -33,10 +33,35 @@ func resourceAccessRule() *schema.Resource {
 				Description: "The permission that the account has on the given directory: read, write or admin",
 			},
 		},
+		Importer: &schema.ResourceImporter{
+			State: resourceAccessRuleImport,
+		},
+	}
+}
+
+func resourceAccessRuleCreate(d *schema.ResourceData, m interface{}) error {
+	provider := m.(providerMeta)
+	client := *provider.client
+
+	path := d.Get("dir_path").(string)
+	permission := d.Get("permission").(string)
+	account := d.Get("account_name").(string)
+
+	_, err := client.AccessRules().Get(path, account)
+	if err == nil {
+		return fmt.Errorf("access rule already exists: %s:%s", path, account)
+	} else if err != api.ErrAccessRuleNotFound {
+		return err
 	}
+
+	_, err = client.AccessRules().Set(path, permission, account)
+
+	d.SetId(path + ":" + account)
+
+	return err
```

Comment:
Usually create does a read call at the end:

`return resourceAccessRuleRead(d, m)`

Reply:
Shall we pick this up in a separate PR? We should change this for the service and service_aws resources as well and it hasn't been introduced in this PR.